Source/swiftlint-dev/Rules+Register.swift

```diff
@@ -74,53 +75,174 @@ struct NewRuleDetails: Equatable {
     }
 }
 
+/// Struct to hold processed rule information and shard calculations
+private struct ProcessedRulesContext {
+    let baseRuleNames: [String]
+    let totalShards: Int
+
+    init(ruleFiles: [String], shardSize: Int) {
+        self.baseRuleNames = ruleFiles.map { $0.replacingOccurrences(of: ".swift", with: "") }
+        guard shardSize > 0, !baseRuleNames.isEmpty else {
+            self.totalShards = baseRuleNames.isEmpty ? 0 : 1
+            return
+        }
+        self.totalShards = (baseRuleNames.count + shardSize - 1) / shardSize // Ceiling division
+    }
+}
+
 private extension SwiftLintDev.Rules.Register {
+    /// Number of test classes per shard for optimal parallelization
+    private static let shardSize = 25
+
+    /// Common parent directory of testsDirectory
+    private var testsParentDirectory: URL {
+        testsDirectory.deletingLastPathComponent()
+    }
+
+    /// Generate content for BuiltInRules.swift file
+    private func generateBuiltInRulesFileContent(rulesImportList: String) -> String {
+        """
+        // GENERATED FILE. DO NOT EDIT!
+
+        /// The rule list containing all available rules built into SwiftLint.
+        public let builtInRules: [any Rule.Type] = [
+        \(rulesImportList.indent(by: 4)),
+        ]
+
+        """
+    }
+
+    /// Generate content for Swift test files
+    private func generateSwiftTestFileContent(forTestClasses testClassesString: String) -> String {
+        """
+        // swiftlint:disable file_length file_name single_test_class type_name
+        // swiftlint:disable:previous blanket_disable_command superfluous_disable_command
+        // GENERATED FILE. DO NOT EDIT!
```

Comment:
Let's have this note/warning first in the file separated from the rest to make it more pronounced.

Reply:
e3b53fe1005351d9ced4a760116a84cb1e4e923b